### PR TITLE
Split up external tests

### PR DESCRIPTION
**Pull request by erak**
Splits up external tests into separate scripts and CircleCI jobs.

### CODE REVIEW

**Review by chriseth — DISMISSED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -101,9 +101,28 @@ jobs:
...
+  test_emscripten_external_gnosis:
+    docker:
+      - image: circleci/node:10
+    environment:
+      TERM: xterm
+    steps:
+      - checkout
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
```
Comment (chriseth) [inline .circleci/config.yml:112]: Can be removed - circle prints the config anyway.

File: .circleci/config.yml [hunk 2]
```diff
@@ -101,9 +101,28 @@ jobs:
...
+          name: Install external tests deps
+          command: |
+            node --version
+            npm --version
+      - run:
+          name: External Zeppelin tests
+          command: |
+            test/externalTests/gnosis-safe.sh /tmp/workspace/soljson.js || test/externalTests/gnosis-safe.sh /tmp/workspace/soljson.js
```
Comment (chriseth) [inline .circleci/config.yml]: Either wrong script or wrong step name.

File: .circleci/config.yml [hunk 3]
```diff
@@ -101,9 +101,28 @@ jobs:
             node --version
             npm --version
       - run:
-          name: External tests
+          name: External GnosisSafe tests
```
Comment (chriseth) [inline .circleci/config.yml:99]:
```suggestion
          name: External Zeppelin tests
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -101,9 +101,28 @@ jobs:
...
+      - checkout
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: Install external tests deps
+          command: |
+            node --version
+            npm --version
+      - run:
+          name: External Zeppelin tests
```
Comment (chriseth) [inline .circleci/config.yml:113]:
```suggestion
          name: External GnosisSafe tests
```

**Comment by erak** [comment]
Updated. @chriseth Could you please change settings, such that `test_emscripten_external` is not required anymore? Does it make sense to add the new jobs temporarily, before we remove them from `build_all` with https://github.com/ethereum/solidity/pull/6424 ?

**Comment by chriseth** [comment]
Changed settings. Sure, let's add the jobs for one PR.

**Review by chriseth — DISMISSED** [review]

File: test/externalTests/shared.sh [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    sed -i s/solc.compileStandard/solc.compile/ "node_modules/truffle/build/cli.bundled.js"
+}
+
+function find_truffle_config
+{
+    local dir="$1"
+    local config_file="truffle.js"
+    local alt_config_file="truffle-config.js"
+
+    if [ ! -f "$dir/$config_file" ] && [ ! -f "$alt_config_file" ]; then
```
Comment (erak) [inline test/externalTests/shared.sh]: This is wrong. Will fix it.

Comment (erak) [reply test/externalTests/shared.sh]: Done.

**Comment by erak** [comment]
I simplified the tests a bit more by adding wrapper functions for project-specific calls and introducing the `OPTIMIZER_LEVEL`.

File: test/externalTests/gnosis.sh [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+function gnosis_safe_test
+{
+    OPTIMIZER_LEVEL=1
+    setup https://github.com/gnosis/safe-contracts.git development
+    run_init "npm install"
+
+    CONFIG=$(find_truffle_config)
+    replace_libsolc_call
+
+    run_test "npx truffle compile" "npm test"
```
Comment (erak) [inline test/externalTests/gnosis.sh]: This does not propagate errors correctly. I will need to rollback to a previous solution:
```
funcition init_fn { npm install; }
funcition compile_fn { npx truffle compile; }
funcition test_fn { npm test; }

run_init init_fn
...
run_test compile_fn test_fn
```

**Comment by stackenbotten** [comment]
There was an error when running `test_check_style` for commit `2308df01c63732ac341c2b4c317a63b35fbbf4c2`:
```
Error: Trailing whitespace found:
 test/externalTests/common.sh:215: 

```
Please check that your changes are working as intended.

**Review by chriseth — APPROVED** [review]

**Pull request merged**